framework/tlvf/AutoGenerated/src/CmduMessageRx.cpp

```diff
@@ -49,7 +250,17 @@ std::shared_ptr<cCmduHeader> CmduMessageRx::parse(uint8_t *buff, size_t buff_len
     m_cmdu_header = std::make_shared<cCmduHeader>(buff, buff_len, true, false);
     if (!m_cmdu_header || m_cmdu_header->isInitialized() == false) {
         m_cmdu_header = nullptr;
+        return false;
+    }
+    
+    if (!parse_tlvs)
+        return true;
+
+    while (auto tlv = parseNextTlv()) {
+        if (std::dynamic_pointer_cast<tlvEndOfMessage>(tlv)) {
```

Comment:
I wonder if the dynamic pointer cast could return success if the TLV is not tlvEndOfMessage. I'm not sure.
I would explicitly compare the tlv type to the tlvEndOfMessage type.

Reply:
This is exactly what is done in `parseNextTlv()`. It will fail if it is not exactly this class, so no need to do any explicit comparisons.